fix: SC2262 shellcheck warning on deeper bash and zsh integration

<!-- Thank you for being interested in nvm! Please help us by filling out the following form if you‘re having trouble. If you have a feature request, or some other question, please feel free to clear out the form. Thanks! -->

#### Operating system and version:

#### `nvm debug` output:
<details>
<!-- do not delete the following blank line -->

```sh
$TERM_PROGRAM: Apple_Terminal
$SHELL: /bin/bash
$SHLVL: 1
whoami: 'djc'
${HOME}: /Users/djc
${NVM_DIR}: '${HOME}/.nvm'
${PATH}: ${HOME}/.pyenv/shims:${HOME}/.pyenv/bin:${HOME}/.rbenv/shims:/opt/homebrew/bin:/opt/homebrew/sbin:~/bin:${NVM_DIR}/versions/node/v16.15.0/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/go/bin:/Library/Apple/usr/bin
$PREFIX: ''
${NPM_CONFIG_PREFIX}: ''
$NVM_NODEJS_ORG_MIRROR: ''
$NVM_IOJS_ORG_MIRROR: ''
shell version: 'GNU bash, version 3.2.57(1)-release (arm64-apple-darwin21)'
uname -a: 'Darwin 21.6.0 Darwin Kernel Version 21.6.0: Thu Sep 29 20:13:56 PDT 2022; root:xnu-8020.240.7~1/RELEASE_ARM64_T6000 arm64'
checksum binary: 'shasum'
OS version: macOS 12.6.1 21G217
curl: /usr/bin/curl, curl 7.79.1 (x86_64-apple-darwin21.0) libcurl/7.79.1 (SecureTransport) LibreSSL/3.3.6 zlib/1.2.11 nghttp2/1.45.1
wget: not found
sed: /usr/bin/sed
cut: /usr/bin/cut
basename: /usr/bin/basename
rm: /bin/rm
mkdir: /bin/mkdir
xargs: /usr/bin/xargs
git: /opt/homebrew/bin/git, git version 2.35.1
grep: /usr/bin/grep, grep (BSD grep, GNU compatible) 2.6.0-FreeBSD
awk: /usr/bin/awk, awk version 20200816
nvm current: v16.15.0
which node: ${NVM_DIR}/versions/node/v16.15.0/bin/node
which iojs: 
which npm: ${NVM_DIR}/versions/node/v16.15.0/bin/npm
npm config get prefix: ${NVM_DIR}/versions/node/v16.15.0
npm root -g: ${NVM_DIR}/versions/node/v16.15.0/lib/node_modules
```
</details>

#### `nvm ls` output:
<details>
<!-- do not delete the following blank line -->

```sh
->     v16.15.0
default -> lts/* (-> v16.15.0)
iojs -> N/A (default)
unstable -> N/A (default)
node -> stable (-> v16.15.0) (default)
stable -> 16.15 (-> v16.15.0) (default)
lts/* -> lts/gallium (-> v16.15.0)
lts/argon -> v4.9.1 (-> N/A)
lts/boron -> v6.17.1 (-> N/A)
lts/carbon -> v8.17.0 (-> N/A)
lts/dubnium -> v10.24.1 (-> N/A)
lts/erbium -> v12.22.12 (-> N/A)
lts/fermium -> v14.19.3 (-> N/A)
lts/gallium -> v16.15.0
```
</details>

#### How did you install `nvm`?
<!-- (e.g. install script in readme, Homebrew) -->
install script in readme

#### What steps did you perform?
Using deeper bash and zsh completion: https://github.com/nvm-sh/nvm#bash

#### What happened?
Running shellcheck, I get this error on a few lines: https://www.shellcheck.net/wiki/SC2262

```sh
# bash
...
alias cd='cdnvm'
cd "$PWD" || exit
...
# zsh
...
local node_version="$(nvm version)"
local nvmrc_path="$(nvm_find_nvmrc)"

if [ -n "$nvmrc_path" ]; then
  local nvmrc_node_version=$(nvm version "$(cat "${nvmrc_path}")")
```

#### What did you expect to happen?
No errors

#### Is there anything in any of your profile files that modifies the `PATH`?
<!--  (e.g. `.bashrc`, `.bash_profile`, `.zshrc`, etc) -->
Bunch of stuff, but don't think it's related.

> #### Operating system and version:
> #### `nvm debug` output:
> #### `nvm ls` output:
> #### How did you install `nvm`?
> install script in readme
> 
> #### What steps did you perform?
> Using deeper bash and zsh completion: https://github.com/nvm-sh/nvm#bash
> 
> #### What happened?
> Running shellcheck, I get this error on a few lines: https://www.shellcheck.net/wiki/SC2262
> 
> ```shell
> # bash
> # ...
> alias cd='cdnvm'
> cd "$PWD" || exit
> # ...
> # zsh
> # ...
> local node_version="$(nvm version)"
> local nvmrc_path="$(nvm_find_nvmrc)"
> 
> if [ -n "$nvmrc_path" ]; then
>   local nvmrc_node_version=$(nvm version "$(cat "${nvmrc_path}")")
> # ...
> ```
> 
> #### What did you expect to happen?
> No errors
> 
> #### Is there anything in any of your profile files that modifies the `PATH`?
> Bunch of stuff, but don't think it's related.

--- a/README.md
+++ b/README.md
@@ -599,8 +599,9 @@ cdnvm() {
         fi
     fi
 }
+
 alias cd='cdnvm'
-cd "$PWD"
+cdnvm "$PWD" || exit
 ```
 
 This alias would search 'up' from your current directory in order to detect a `.nvmrc` file. If it finds it, it will switch to that version; if not, it will use the default version.
@@ -615,11 +616,14 @@ Put this into your `$HOME/.zshrc` to call `nvm use` automatically whenever you e
 ```zsh
 # place this after nvm initialization!
 autoload -U add-zsh-hook
+
 load-nvmrc() {
-  local nvmrc_path="$(nvm_find_nvmrc)"
+  local nvmrc_path
+  nvmrc_path="$(nvm_find_nvmrc)"
 
   if [ -n "$nvmrc_path" ]; then
-    local nvmrc_node_version=$(nvm version "$(cat "${nvmrc_path}")")
+    local nvmrc_node_version
+    nvmrc_node_version=$(nvm version "$(cat "${nvmrc_path}")")
 
     if [ "$nvmrc_node_version" = "N/A" ]; then
       nvm install
@@ -631,6 +635,7 @@ load-nvmrc() {
     nvm use default
   fi
 }
+
 add-zsh-hook chpwd load-nvmrc
 load-nvmrc
 ```

